resources/assets/lib/user-group-badge.tsx

```diff
@@ -15,28 +16,35 @@ export default function UserGroupBadge({group, modifiers}: Props) {
     return null;
   }
 
-  const style = osu.groupColour(group);
-
-  let blockClass = classWithModifiers('user-group-badge', {
-    probationary: group.is_probationary,
-    [group.identifier]: true,
-  });
-  blockClass += classWithModifiers('user-group-badge', modifiers, true);
-
-  const playModes: JSX.Element[] = (group.playmodes ?? []).map((mode) => <i key={mode} className={`fal fa-extra-mode-${mode}`} />);
-
-  return (
-    <div
-      className={blockClass}
-      data-label={group.short_name}
-      style={style}
-      title={group.name}
-    >
-      {playModes.length > 0 &&
-        <div className={'user-group-badge__modes'}>
-          {playModes}
-        </div>
-      }
-    </div>
-  );
+  let title = group.name;
+
+  if (group.playmodes != null && group.playmodes.length > 0) {
+    const playmodeNames = group.playmodes
+      .map((playmode) => osu.trans(`beatmaps.mode.${playmode}`))
+      .join(', ');
+
+    title += ` (${playmodeNames})`;
+  }
+
+  const props = {
+    'children': group.playmodes != null && group.playmodes.length > 0 && (
+      <div className={'user-group-badge__modes'}>
+        {group.playmodes.map((playmode) => (
+          <i key={playmode} className={`fal fa-extra-mode-${playmode}`} />
+        ))}
+      </div>
+    ),
+    'className': classWithModifiers('user-group-badge', {
+      probationary: group.is_probationary,
+      [group.identifier]: true,
+      ...modifiers,
+    }),
+    'data-label': group.short_name,
+    'style': osu.groupColour(group),
+    title,
+  };
+
+  return group.id < 0
```

Comment:
where does this negative group id come from?

Reply:
mapper and guest group

Reply:
ok actually I just remembered this was the whole reason I added `has_listing` to groups a while ago... will use that instead and  update blade component too